Fixes #SCAN4NET-423

pieterkl opened the pull request:
<!-- 
  Only for standalone PRs without Jira issue in the PR title: 
    * Replace this comment with Epic ID to create a new Task in Jira
    * Replace this comment with Issue ID to create a new Sub-Task in Jira
    * Ignore or delete this note to create a new Task in Jira without a parent 
-->

alex-meseldzija-sonarsource commented on src/SonarScanner.MSBuild.TFS/TrxFileReader.cs:
```diff
@@ -213,20 +219,21 @@ private Dictionary<string, List<string>> GetCoverageAttachments(IEnumerable<stri
-                // With VSTest task the coverage file name uses underscore instead of spaces.
-                Path.Combine(trxDirectoryName, trxFileName.Replace(' ', '_'), "In", attachmentUri)
-            };
+        var possibleCoveragePaths = new List<string>();
```
We can keep the mandatory values within the initialisation of the List
```csharp
  var possibleCoveragePaths = new List<string>()
  {
      attachmentUri,
      Path.Combine(trxDirectoryName, trxFileName, "In", attachmentUri),
      Path.Combine(trxDirectoryName, trxFileName.Replace(' ', '_'), attachmentUri),
      // https://jira.sonarsource.com/browse/SONARMSBRU-361
      // With VSTest task the coverage file name uses underscore instead of spaces.
      Path.Combine(trxDirectoryName, trxFileName.Replace(' ', '_'), "In", attachmentUri),
  };
```

pieterkl replied:
Good suggestion, i've implemented this in the new commit for this pull request.

alex-meseldzija-sonarsource reviewed (changes requested):
Thanks for the contribution!!

I have a few suggestions and we do definitely need tests.

We already cover cases with no `runDeploymentRootNode` in our UTs, so right now I can think of:

- `runDeploymentRootNode` exists, but has no `runDeploymentRoot` attribute
- `runDeploymentRootNode` exists, but `runDeploymentRoot` attribute is an empty string
- `runDeploymentRootNode` exists, `runDeploymentRoot` is a valid string (the happy path)

Tests should be in [TrxFileReaderTests.cs](https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/Tests/SonarScanner.MSBuild.TFS.Test/TrxFileReaderTests.cs) and can likely follow a similar format. 

You will probably need to update the [GetTrxContent](https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/Tests/SonarScanner.MSBuild.TFS.Test/TrxFileReaderTests.cs#L313) method in that class, or construct your own to fully test these changes.

If you need or would like any help/guidance feel free to reach out.

pieterkl commented:
I've used your suggestions to update the PR and added some unit tests. All other tests already test trx files without a runDeploymentRoot, so i did not add an extra test for this situation. I've added a test however that checks how the code handles a trx file with a wrong path.

alex-meseldzija-sonarsource commented:
Hi @pieterkl, apologies for the delay.

To proceed with your contribution, we need to ask you whether you're okay with signing a Contributor License Agreement (CLA)?
This is a standard procedure for external contributions at sonar.

If yes, could you provide your email so someone from our legal team can contact you for the CLA.
(you can email me at the email in my profile).
Thanks!

alex-meseldzija-sonarsource commented:
Hi @pieterkl We will need a response within the next 2 weeks otherwise we will need to close this PR and implement it ourselves. Thanks
